src/txn/txn_recover.c

```diff
@@ -974,8 +974,8 @@ __wt_txn_recover(WT_SESSION_IMPL *session, const char *cfg[])
          */
         WT_ERR(session->iface.checkpoint(&session->iface, "force=1"));
 
-    /* Initialize the connection's base write generation after rollback to stable. */
-    WT_ERR(__wt_metadata_init_base_write_gen(session));
+    /* Update the connection's base write generation after rollback to stable. */
+    WT_ERR(__wt_metadata_update_all_base_write_gen(session));
```

Comment:
What case is this covering? The PR description is vague.

Reply:
The earlier code is reinitialized the connection base write gen to 1 and update with the metadata checkpoint write generation number. It is possible that the other files may have more write gen updates than the metadata, so the connection base write gen must get updated with the latest write gen from all the files.

In the case of backup target scenarios, the connection base write gen will be reset to 1 is the main problem due to which the test_backup03 was failing. 

I will add some more comments to the above point to be clear of why we are updating the connection base write gen from all the files.

Reply:
Added a comment to explain the need for that function. Can you check it again?